### PR TITLE
Add support for TaxId resource and APIs

### PR DESCRIPTION
**Pull request by remi-stripe**
cc @stripe/api-libraries

### CODE REVIEW

**Comment by remi-stripe** [comment]
This is ready to be reviewed. Assigning to @mickjermsurawong-stripe to compare with stripe-java

File: src/Stripe.net/Entities/TaxIds/TaxId.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        /// Value of the tax ID.
+        /// </summary>
+        [JsonProperty("value")]
+        public string Value { get; set; }
+
+        /// <summary>
+        /// The persons’s verification status.
+        /// </summary>
+        [JsonProperty("verification")]
+        public PersonVerification Verification { get; set; }
```
Comment (mickjermsurawong-stripe) [inline src/Stripe.net/Entities/TaxIds/TaxId.cs]: Should this be `TaxIdVerification`?

File: src/Stripe.net/Entities/Customers/Customer.cs [hunk 2]
```diff
@@ -164,15 +164,23 @@ internal object InternalDefaultSource
         [JsonProperty("subscriptions")]
         public StripeList<Subscription> Subscriptions { get; set; }
 
+        /// <summary>
+        /// The customer’s current tax ids, if any.
+        /// </summary>
+        [JsonProperty("tax_ids")]
+        public StripeList<TaxId> TaxIds { get; set; }
```
Comment (mickjermsurawong-stripe) [inline src/Stripe.net/Entities/Customers/Customer.cs]: Should we also update `CustomerTaxIdOptions`? https://github.com/stripe/stripe-dotnet/tree/master/src/Stripe.net/Services/Customers
https://site-admin.stripe.com/docs/api/customers/create#create_customer-tax_id_data

**Comment by mickjermsurawong-stripe** [comment]
ptal @remi-stripe

**Comment by remi-stripe** [comment]
Okay, added missing `tax_id_data` and also made sure to deprecate all TaxInfo* related resource/params.

@mickjermsurawong-stripe PTAL
@ob-stripe Can you review too as it's a complex resource/change?

**Comment by remi-stripe** [comment]
@ob-stripe @mickjermsurawong-stripe Friendly bump as this is going out now in the API

File: src/Stripe.net/Entities/Invoices/InvoiceCustomerTaxId.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+namespace Stripe
+{
+    using Newtonsoft.Json;
+    using Stripe.Infrastructure;
+
+    public class InvoiceCustomerTaxId : StripeEntity
+    {
+        [JsonProperty("type")]
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/InvoiceCustomerTaxId.cs]: Mind adding documentation here?
```c#
/// <summary>
/// The type of the tax ID, one of <c>eu_vat</c>, <c>nz_gst</c>, <c>au_abn</c>,
//// or <c>unknown</c>
/// </summary>
```

File: src/Stripe.net/Entities/Invoices/InvoiceCustomerTaxId.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        public string Type { get; set; }
+
+        [JsonProperty("value")]
```
Comment (ob-stripe) [inline src/Stripe.net/Entities/Invoices/InvoiceCustomerTaxId.cs]: And here:
```c#
/// <summary>The value of the tax ID.</summary>
```

**Comment by remi-stripe** [comment]
@ob-stripe Fixed, PTAL

**Pull request merged**